### ISSUE TITLE
phpdoc_types_order messes up intersection type in Docblock

### ISSUE THREAD

**Issue by 7ochem** [FriendsOfPHP/PHP-CS-Fixer#6009]
## Bug report

I just updated PHP CS Fixer from 2.19.0 to 2.19.2. In 2.19.1 there are a lot of fixes and one of those fixes now has a strange effect on an intersection type in a Docblock.

I'm running PHP CS FIxer 2.19.0 vs. 2.19.2 on **PHP 7.4.23** with the command `php-cs-fixer fix --dry-run --diff -vvv` and the following configuration file:
<details>
  <summary>.php-cs-fixer.dist.php</summary>

  ```php
  <?php declare(strict_types=1);
  
  $finder = PhpCsFixer\Finder::create()
      ->in(__DIR__ . '/src')
      ->notPath('Kernel.php')
  ;
  
  return (new PhpCsFixer\Config())
      ->setRiskyAllowed(true)
      ->setRules([
          '@Symfony' => true,
          'blank_line_after_opening_tag' => false,
          'blank_line_before_statement' => false,
          'cast_spaces' => ['space' => 'none'],
          'class_attributes_separation' => ['elements' => ['method' => 'one', 'property' => 'one']],
          'comment_to_phpdoc' => [],
          'concat_space' => ['spacing' => 'one'],
          'declare_strict_types' => true,
          'linebreak_after_opening_tag' => false,
          'no_superfluous_phpdoc_tags' => ['allow_mixed' => true, 'allow_unused_params' => false],
          'ordered_class_elements' => ['order' => ['use_trait', 'constant', 'property', 'construct']],
          'phpdoc_add_missing_param_annotation' => ['only_untyped' => true],
          'phpdoc_align' => false,
          'phpdoc_annotation_without_dot' => false,
          'phpdoc_separation' => false,
          'phpdoc_summary' => false,
          'yoda_style' => null,
      ])
      ->setFinder($finder)
      ->setCacheFile('.php-cs-fixer.cache') // forward compatibility with 3.x line
  ;
  ```
</details>

I now get the following output (suggested change) for my code.

```diff
   1) src/Admin/FooAdmin.php (phpdoc_types_order)
      ---------- begin diff ----------
--- Original
+++ New
@@ @@
 /**
- * @method ModelManagerInterface<Foo>&ModelManager<Foo> getModelManager()
+ * @method ModelManagerInterface<Foo>&ModelManager|<|Foo> getModelManager()
  */
 class FooAdmin extends Admin
```

### Code snippet that reproduces the problem

```php
<?php declare(strict_types=1);

namespace App\Admin;

use App\Entity\Foo;
use Sonata\AdminBundle\Admin\AbstractAdmin;
use Sonata\AdminBundle\Model\ModelManagerInterface;
use Sonata\DoctrineORMAdminBundle\Model\ModelManager;

/**
 * @method ModelManagerInterface<Foo>&ModelManager<Foo> getModelManager()
 */
class FooAdmin extends AbstractAdmin
{
}
```

**Comment by SpacePossum** [comment]
thanks for reporting,

linking to the meta [PHP8.1](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5891) issue here

**Comment by 7ochem** [comment]
> linking to the meta [PHP8.1](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5891) issue here

Thanks. This is in PHP 7.4 in a Docblock though.

**Comment by oliverklee** [comment]
Possible a duplicate of #5847/#5798.

**Comment by 7ochem** [comment]
> Possible a duplicate of #5847/#5798.

Those are all closed. There has been some fix in 3.0.2, but this is in 2.19.1

This is probably a duplicate of #5850. That one is still open

**Comment by julienfalque** [comment]
Version 2.19 is not maintained anymore. This is likely the same issue as #5850.
